src/v/archival/ntp_archiver_service.cc

```diff
@@ -2160,7 +2160,7 @@ ss::future<> ntp_archiver::garbage_collect_archive() {
         throw std::system_error(backlog.error());
     }
 
-    std::deque<std::filesystem::path> objects_to_remove;
+    std::vector<lw_segment_meta> segments_to_remove;
```

Comment:
Seastar statically preallocates all of its memory when the application starts. For long running apps (like Redpanda), this leads to fragmentation issues. For this reason, we generally favour data structures which don't require big chunks of contiguous memory (in this case `deque` was chosen over `vector`).

Travis wrote up a great [internal wiki](https://vectorizedio.atlassian.net/wiki/spaces/CORE/pages/318275653/Memory+Management+in+Redpanda) on this topic.